### PR TITLE
GLOCKFILE: update dependencies

### PR DESCRIPTION
**Pull request by tamird**
The most notable change here is a monstrous update to go-staticcheck, which is *amazing*. A bunch of tests here were rewritten to avoid calling `(*testing.T).Fatal{,f}` in a non-main goroutine (since this is now detected by staticcheck), so please take a look.

cc @spencerkimball @petermattis 

Summary of not-definitely-irrelevant changes:
- github.com/chzyer/readline:
	- fixed panic when entering "setprompt": https://github.com/chzyer/readline/commit/c914be6
- github.com/golang/lint:
	- flag error strings that end with a newline: https://github.com/golang/lint/commit/206c0f0
- honnef.co/go/staticcheck:
	- too many to list, replaces ineffassign: https://github.com/dominikh/go-staticcheck/compare/02337b3...3aa60d6

Skipped:
- github.com/cockroachdb/c-rocksdb (#9616)
- github.com/docker/docker (https://github.com/docker/docker/pull/27912) breaks all dependents
- google.golang.org/grpc (#9697)

---
This change is [Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/10711)

### CODE REVIEW

**Comment by bdarnell** [comment]
:lgtm:

---

Reviewed 17 of 17 files at r1.
Review status: all files reviewed at latest revision, 5 unresolved discussions, all commit checks successful.

---

*[build/style_test.go, line 713 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/10711#-KWfm403HlznYfS7ePK3:-KWfm403HlznYfS7ePK4:bj36vie) ([raw file](https://github.com/cockroachdb/cockroach/blob/066862bae3b3030ce9e488dc49af57ff05b04b05/build/style_test.go#L713)):*
> ```Go
> 			strings.Join([]string{
> 				"github.com/cockroachdb/cockroach/pkg/sql/parser/sql.y:SA4006",
> 				"github.com/cockroachdb/cockroach/pkg/sql/errors.go:SA4004",
> ```

Add a comment defining the error codes we're ignoring.

---

*[pkg/cli/dump.go, line 99 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/10711#-KWfmG_WJ8wRS7h9KqlZ:-KWfmG_WJ8wRS7h9Kql_:b-6xxk85) ([raw file](https://github.com/cockroachdb/cockroach/blob/066862bae3b3030ce9e488dc49af57ff05b04b05/pkg/cli/dump.go#L99)):*
> ```Go
> 			return err
> 		}
> 		nameI, typI := vals[0], vals[1]
> ```

What rule is being enforced here?

---

*[pkg/sql/prepare.go, line 142 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/10711#-KWfmZ8LwGLi464Y0ZYz:-KWfmZ8LwGLi464Y0ZZ-:b-eoh8uj) ([raw file](https://github.com/cockroachdb/cockroach/blob/066862bae3b3030ce9e488dc49af57ff05b04b05/pkg/sql/prepare.go#L142)):*
> ```Go
> // This is used by the "delete" message in the pgwire protocol; after DeleteAll
> // statements and portals can be added again.
> func (ps *PreparedStatements) DeleteAll() {
> ```

Nice that we now have something that can detect this bug. 

---

*[pkg/storage/intent_resolver.go, line 401 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/10711#-KWfn3ShnpaBbimpkg0t:-KWfn3SiudN92DZnz753:bxk639p) ([raw file](https://github.com/cockroachdb/cockroach/blob/066862bae3b3030ce9e488dc49af57ff05b04b05/pkg/storage/intent_resolver.go#L401)):*
> ```Go
> // the same intents again (in the absence of #8360, we provide this
> // guarantee by resolving the intents synchronously regardless of the
> // `wait` argument).
> ```

I'm not a fan of this rule - if we can't name the argument then it's harder to talk about (as in this comment). 

---

*[pkg/storage/replica.go, line 3304 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/10711#-KWfnk3KV2GeSmx91PgL:-KWfnk3KV2GeSmx91PgM:bkwg18i) ([raw file](https://github.com/cockroachdb/cockroach/blob/066862bae3b3030ce9e488dc49af57ff05b04b05/pkg/storage/replica.go#L3304)):*
> ```Go
> 	writeBatch *storagebase.WriteBatch,
> ) (enginepb.MVCCStats, *roachpb.Error) {
> 	if rpd.State.RaftAppliedIndex == 0 {
> ```

Also not sure about this one - it's encoding knowledge of RaftAppliedIndex's type here; if we were to change this to a signed int (in keeping with our general preference for signed types), this would no longer be enforcing the same invariant as before. I'm not especially afraid of negative numbers showing up in this context even if we did change the type, but the rule seems questionable to me. 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/10711#-:-KWfoiaD-FrdmE5dqhSe:bnfp4nl)*

**Comment by tamird** [comment]
Review status: 13 of 16 files reviewed at latest revision, 5 unresolved discussions.

---

*[build/style_test.go, line 713 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/10711#-KWfm403HlznYfS7ePK3:-KWfqcHH-NgmGr3GOg5W:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/066862bae3b3030ce9e488dc49af57ff05b04b05/build/style_test.go#L713)):*
<details><summary><i>Previously, bdarnell (Ben Darnell) wrote…</i></summary>
> Add a comment defining the error codes we're ignoring.
</details>
Done.

---

*[pkg/cli/dump.go, line 99 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/10711#-KWfmG_WJ8wRS7h9KqlZ:-KWfqh8KuGMhRFhXkS6s:b-o2jij1) ([raw file](https://github.com/cockroachdb/cockroach/blob/066862bae3b3030ce9e488dc49af57ff05b04b05/pkg/cli/dump.go#L99)):*
<details><summary><i>Previously, bdarnell (Ben Darnell) wrote…</i></summary>
> What rule is being enforced here?
</details>
Out of bounds access. Note that the old code had `vals[4]` in the error message. The extraction of local variables is just for my own sanity.

---

*[pkg/sql/prepare.go, line 142 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/10711#-KWfmZ8LwGLi464Y0ZYz:-KWfqpEfBk0F4C9MseTx:bb77uci) ([raw file](https://github.com/cockroachdb/cockroach/blob/066862bae3b3030ce9e488dc49af57ff05b04b05/pkg/sql/prepare.go#L142)):*
<details><summary><i>Previously, bdarnell (Ben Darnell) wrote…</i></summary>
> Nice that we now have something that can detect this bug. 
</details>
Yep!

---

*[pkg/storage/intent_resolver.go, line 401 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/10711#-KWfn3ShnpaBbimpkg0t:-KWfrJR0_wK5F1PPP-PK:bp3aow6) ([raw file](https://github.com/cockroachdb/cockroach/blob/066862bae3b3030ce9e488dc49af57ff05b04b05/pkg/storage/intent_resolver.go#L401)):*
<details><summary><i>Previously, bdarnell (Ben Darnell) wrote…</i></summary>
> I'm not a fan of this rule - if we can't name the argument then it's harder to talk about (as in this comment). 
</details>
Reverted.

---

*[pkg/storage/replica.go, line 3304 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/10711#-KWfnk3KV2GeSmx91PgL:-KWfre-zoGYDff9tWqrw:bz1bc62) ([raw file](https://github.com/cockroachdb/cockroach/blob/066862bae3b3030ce9e488dc49af57ff05b04b05/pkg/storage/replica.go#L3304)):*
<details><summary><i>Previously, bdarnell (Ben Darnell) wrote…</i></summary>
> Also not sure about this one - it's encoding knowledge of RaftAppliedIndex's type here; if we were to change this to a signed int (in keeping with our general preference for signed types), this would no longer be enforcing the same invariant as before. I'm not especially afraid of negative numbers showing up in this context even if we did change the type, but the rule seems questionable to me. 
</details>
Agreed, reverted.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/10711)*

**Review by petermattis — APPROVED** [review]
LGTM

File: GLOCKFILE [hunk 1]
```diff
@@ -78,44 +75,44 @@ github.com/kr/pretty cfb55aafdaf3ec08f0db22699ab822c50091b1c4
 github.com/kr/text 7cafcd837844e784b526369c9bce262804aebc60
 github.com/leekchan/timeutil 28917288c48df3d2c1cfe468c273e0b2adda0aa5
 github.com/lib/pq d8eeeb8bae8896dd8e1b7e514ab0d396c4f12a1b
-github.com/lightstep/lightstep-tracer-go 91ae78ce073799b6a40e303d7dec6cc3a4257915
+github.com/lightstep/lightstep-tracer-go b1bc6c278e9a007584bbc0b9139e776b4e3203e3
```
Comment (dt) [inline GLOCKFILE]: Did you intentionally pick up https://github.com/lightstep/lightstep-tracer-go/pull/58 here ?

**Comment by tamird** [comment]
Uh, yes? Why is that PR remarkable for us?

**Pull request merged**